components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -3946,10 +3946,67 @@ private void associateAuthenticationHistory(SessionDataCacheEntry resultFromLogi
             for (AuthHistory authHistory : sessionContext.getSessionAuthHistory().getHistory()) {
                 authMethods.add(authHistory.toTranslatableString());
             }
+            authMethods = getAMRValues(authMethods, sessionContext.getAuthenticatedIdPs());
             resultFromLogin.getParamMap().put(OAuthConstants.AMR, authMethods.toArray(new String[authMethods.size()]));
         }
     }
 
+    /**
+     * Replaces the authenticator names with the AMR values sent by the IDP.
+     *
+     * @param authMethods       The list of authentication methods set by resident IDP.
+     * @param authenticatedIdPs The authenticated IDPs.
+     */
+    private List<String> getAMRValues(List<String> authMethods, Map<String, org.wso2.carbon.identity
+            .application.authentication.framework.model.AuthenticatedIdPData> authenticatedIdPs) {
```

Comment:
can't we import and use AuthenticatedIdPData?

Reply:
Fixed with 986056862998a05be12dba4c5697d91175210a9a